plugins/woocommerce-blocks/tests/e2e/utils/types.ts

```diff
@@ -1 +1,6 @@
-export type TemplateType = 'wp_template' | 'wp_template_part';
+export type BlockData< T = unknown > = {
+	name: string;
+	slug: string;
+	mainClass: string;
+	selectors: Record< 'editor' | 'frontend', Record< string, unknown > >;
+} & ( T extends undefined ? Record< string, never > : T );
```

Comment:
When I created this type, the idea was to "enforce" folks to keep the same style when they define the object for store selectors/slug, but this type isn't used so much. Do you think that we should keep it? I'm fine with removing it.

Reply:
IMO, it's currently used enough to justify a dedicated export (31 instances across 15 files). However, I agree that it's not very useful as it adds a layer of complexity to the tests and impairs the code readability. It's generally better to repeat a string a couple of times than to interpolate with a constant, e.g.:

```js
test.describe( `${ blockData.name } Block`, () => {
  // ...
} );
```

Therefore, I think we can drop the `BlockData` convention! 😄 I'm not sure, though, if this PR is a good place to do that, as it's already massive. Should we do it in a follow-up PR?

Reply:
> Should we do it in a follow-up PR? 

Yeah, sure! 🙇